File: applications/event-account-jdbc-sink/src/main/java/com/vmware/account/jdbc/sink/consumer/AccountConsumer.java
package com.vmware.account.jdbc.sink.consumer;

import com.vmware.account.jdbc.sink.domain.AccountEntity;
import com.vmware.account.jdbc.sink.repository.AccountRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

import java.util.function.Consumer;

    @Component
    @RequiredArgsConstructor
    @Slf4j
    public class AccountConsumer implements Consumer<AccountEntity> {

        private final AccountRepository repository;
        @Override
        public void accept(AccountEntity account) {
            repository.save(account);

        }
    }

File: applications/stream-account-geode-sink/src/main/kotlin/com/vmware/financial/open/banking/account/consumer/AccountConsumer.kt
package com.vmware.financial.open.banking.account.consumer

import com.vmware.financial.open.banking.account.domain.Account
import com.vmware.financial.open.banking.account.service.AccountService
import org.springframework.stereotype.Component
import java.util.function.Consumer

/**
 * @author Gregory Green
 */
@Component
class AccountConsumer(private val accountService: AccountService) :
    Consumer<Account> {
    override fun accept(account: Account) {
        accountService.createAccount(account)
    }
}
